code/game/atoms_movable.dm

```diff
@@ -714,13 +714,20 @@
 	if(crush_dir == NO_CRUSH_DIR)
 		crush_dir = get_dir(get_turf(src), target_turf)
 
+	var/has_tried_to_move = FALSE
+
 	if(is_blocked_turf(target_turf, TRUE))
-		visible_message("<span class='warning'>[src] seems to rock, but doesn't fall over!</span>")
+		var/turf/orig_turf = get_turf(src)
 		Move(target_turf, get_dir(get_turf(src), target_turf))
```

Comment:
I believe Move here will return false if the mob stayed on the same tile, maybe you should try using that?